loleaflet/src/core/Socket.js

```diff
@@ -268,6 +268,30 @@ L.Socket = L.Class.extend({
 			     'background:#ddf;color:black', color, 'color:black');
 	},
 
+	// The problem: if we process one websocket message at a time, the
+	// browser -loves- to trigger a re-render as we hit the main-loop,
+	// this takes ~200ms on a large screen, and worse we get
+	// producer/consumer issues that can fill a multi-second long
+	// buffer of web-socket messages in the client that we can't
+	// process so - slurp and the emit at idle - its faster to delay!
+	_slurpMessage: function(e) {
+		var that = this;
+		if (!this._slurpQueue || !this._slurpQueue.length) {
+			setTimeout(function() {
+				console.log2('Slurp events ' + that._slurpQueue.length);
+				for (var i = 0; i < that._slurpQueue.length; ++i) {
+					// it is - are you ?
+					that._onMessage(that._slurpQueue[i]);
+				}
+				that._slurpQueue = [];
+			}, 1 /* ms */);
+			that._slurpQueue = [];
+		}
+		// I imagine e doesn't hang around at all nicely and this fails
+		// but lets see ...
+		that._slurpQueue.push(e);
```

Comment:
Shouldn't we check for null or undefined _slurpQueue?

Reply:
if it is null - we took the branch above and initialized it just 1 line above =) I should probably remove the pessimistic comment too I guess =)